mods/git.py

```diff
@@ -141,19 +141,33 @@ def prepare_message_hook(self, request, message, detailed):
                 git_url = message.get_property("git.url")
                 git_repo = message.get_property("git.repo")
                 git_tag = message.get_property("git.tag")
-                message.status_tags.append({
-                    "url": git_url,
-                    "title": format_html("Applied as tag {} in repo {}", git_tag, git_repo),
-                    "type": "info",
-                    "char": "G",
+                if git_url:
+                    message.status_tags.append({
+                        "url": git_url,
+                        "title": format_html("Applied as tag {} in repo {}", git_tag, git_repo),
+                        "type": "info",
+                        "char": "G",
+                        })
+                    if git_repo and git_tag:
+                        message.extra_status.append({
+                            "kind": "good",
+                            "html": format_html('Patches applied successfully (<a href="{}">tree</a>, {}).<br/><samp>git fetch {} {}</samp>',
+                                                git_url, colorbox_a, git_repo, git_tag),
+                            "extra": colorbox_div,
+                            "id": "gitlog"
+                        })
+                else:
+                    message.status_tags.append({
+                        "title": format_html("Patch apply succeeded"),
+                        "type": "info",
+                        "char": "G",
+                        })
+                    message.extra_status.append({
+                        "kind": "good",
+                        "html": format_html('Patches applied successfully'),
```

Comment:
Please link the log here, like `format_html('Patches applied successfully ({})' , colorbox_a)`.